src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -293,6 +291,9 @@ private void ProcessEndRequest(object sender)
 			var context = application.Context;
 			var response = context.Response;
 
+			if (SoapRequest.TryExtractSoapAction(_logger, context.Request, out var soapAction))
+				transaction.Name += $" {soapAction}";
```

Comment:
There is an edge case we should consider here: transactions can be renamed by users via the public API - super rare, but that's possible (just do `Agent.Tracer.CurrentTransaction.Name = "foo"`). If a custom transaction name is used, I think we should not add the SOAP action name. We can do something like that:

```
if (transaction is Transaction realTransaction)
{
	if(!realTransaction.HasCustomName)
		transaction.Name += $" {soapAction}";
}
else // If it's some other implementation of `ITransaction`, let's just append the soap action name
	transaction.Name += $" {soapAction}";
```

Parsing the `soapAction` name can be also saved if we know the transaction has a custom name. Also, I see there is already a casting to `Transaction` going on later in this method - maybe just do the check+parsing there, if possible.

Reply:
Thanks, very good point. I was not aware of that.